src/platform/Linux/CHIPBluezHelper.cpp

```diff
@@ -2008,6 +2008,104 @@ CHIP_ERROR ConnectDevice(BluezDevice1 * apDevice)
     return error;
 }
 
+AdapterIterator::~AdapterIterator()
+{
+    g_object_unref(mManager);
+    g_list_free_full(mObjectList, g_object_unref);
+}
+
+void AdapterIterator::Initialize()
+{
+    GError * error = nullptr;
+
+    mManager = g_dbus_object_manager_client_new_for_bus_sync(G_BUS_TYPE_SYSTEM, G_DBUS_OBJECT_MANAGER_CLIENT_FLAGS_NONE,
+                                                             BLUEZ_INTERFACE, "/", bluez_object_manager_client_get_proxy_type,
+                                                             nullptr /* unused user data in the Proxy Type Func */,
+                                                             nullptr /*destroy notify */, nullptr /* cancellable */, &error);
+
+    VerifyOrExit(mManager != nullptr, ChipLogError(DeviceLayer, "Failed to get DBUS object manager for listing adapters."));
+
+    mObjectList      = g_dbus_object_manager_get_objects(mManager);
+    mCurrentListItem = mObjectList;
+
+exit:
+    if (error != nullptr)
+    {
+        ChipLogError(DeviceLayer, "DBus error: %s", error->message);
+        g_error_free(error);
+    }
+}
+
+template <size_t N>
+void safeNullTerminatedCopy(char (&dest)[N], const char * src)
```

Comment:
Might be worth putting something like this in a general utility.... it's come up a few times in code reviews that we have nothing like `strlcpy`.

Reply:
Done. We already had asomething in CHIPMemString.h, added the template there.